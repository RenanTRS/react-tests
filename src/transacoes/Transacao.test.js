import React from 'react'
import {render} from '@testing-library/react'
import Transacao from './Transacao'
import Transacoes from './Transacoes'

describe('Componente de transação do extrato', ()=>{
    it('O Snapshot do componente deve permanecer sempre o mesmo', () => {

        const {container} = render(<Transacao data="08/09/2020" tipo="saque" valor="20.00"/>)
        expect(container.firstChild).toMatchSnapshot()
    })

    describe('Componente transações', () => {
        it('O snapshot do componente deve sempre renderizar', () => {

            const transacoes = [
                {
                    id: 1,
                    transacao: "saque",
                    valor: "15.00",
                    data: "08/09/2020",
                },
                {
                    id: 2,
                    transacao: "desposito",
                    valor: "20.00",
                    data: "09/09/2021"
                }
            ]
            const {container} = render(<Transacoes transacoes={transacoes} />)
            expect(container.firstChild).toMatchSnapshot()
        })
    })
})