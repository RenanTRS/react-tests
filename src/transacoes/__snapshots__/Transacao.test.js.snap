// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de transação do extrato Componente transações O snapshot do componente deve sempre renderizar 1`] = `
<div
  data-testid="transacoes"
>
  <div
    class="transacao-container"
  >
    <p>
      08/09/2020
    </p>
    <p>
      saque
    </p>
    <p>
      R$ 
      15.00
    </p>
  </div>
  <div
    class="transacao-container"
  >
    <p>
      09/09/2021
    </p>
    <p>
      desposito
    </p>
    <p>
      R$ 
      20.00
    </p>
  </div>
</div>
`;

exports[`Componente de transação do extrato O Snapshot do componente deve permanecer sempre o mesmo 1`] = `
<div
  class="transacao-container"
>
  <p>
    08/09/2020
  </p>
  <p>
    saque
  </p>
  <p>
    R$ 
    20.00
  </p>
</div>
`;
